### PR TITLE
V2.7.1

### LINKED ISSUE #11: BAPS Controller V1 Enabled crashes if COM1 doesn't exist.
The server service infinitely reboots (stuck in Starting state) because it can't find the COM port.

Maybe this should still be logged in event viewer, but not using a terminal exception.

### PR DESCRIPTION
**Pull request by mstratford**
### Bug fixes
 * Fix library tracks not being playable after server restart.
The location field in the sql statement is never filled. Use the function to generate it to be saved in the XML file, so that it can replay it after server restart.
 * Tracks and Artists are now sent over BAPSNET using UTF8. (Fixes: #1)
 * Timers are enabled by default (Fixes: #7)
 * Config menu now updates available audio devices, and channels will switch audio device on new file load.
 * Fixes #11, allows COM Port to be changed, and port list updates on opening config window.

### CODE REVIEW

File: BAPSServerAssembly/BAPSController.h [hunk 1]
```diff
@@ -207,6 +207,13 @@ namespace BAPSServerAssembly
 				{
 					System::Byte data = serialPort->ReadByte();
 					handled=false;
+					if (data == 255)
```
Comment (LordAro) [inline BAPSServerAssembly/BAPSController.h]: 0xff?

**Comment by LordAro** [comment]
It'd be really really nice to fix #20 first, but otherwise this seems done

File: BAPSServerAssembly/BAPSController.h (hunk 1 above)
Comment (mstratford) [reply BAPSServerAssembly/BAPSController.h]: Hmm, I left it as 255 because all the rest of the button codes are in the UI as ints.

**Comment by mstratford** [comment]
I have a minor niggle about the rendering of the config menu to fix and yes, the whole #20 issue.

**Pull request merged**